lib/project_types/extension/commands/serve.rb

```diff
@@ -63,13 +63,7 @@ def start_tunnel_if_required(runtime_configuration)
         return runtime_configuration unless specification_handler.establish_tunnel?(@ctx)
         return runtime_configuration unless runtime_configuration.tunnel_requested?
 
-        return start_tunnel(runtime_configuration) if can_start_tunnel?(runtime_configuration)
-        @ctx.abort(@ctx.message("serve.tunnel_already_running"))
-      end
-
-      def can_start_tunnel?(runtime_configuration)
-        return true if ShopifyCLI::Tunnel.urls.empty?
-        ShopifyCLI::Tunnel.running_on?(runtime_configuration.port)
```

Comment:
It looks like it was possible to create a tunnel on a different port when you already had another one running, although only with `extension serve`, and the `tunnel` commands didn't deal with that. Is this something that we should keep?

Right now, only one tunnel would be allowed at the same time, but we could also support it.

Reply:
> It looks like it was possible to create a tunnel on a different port when you already had another one running

Hm, I'd imagine that'd be pretty important for extension development workflows - e.g. for cases where you're working on an UI extension + your app home at the same time, or working on two UI extensions at the same time.

Reply:
It wouldn't be difficult. But I think we should also improve start/stop/status commands to work with multiple tunnels as well. Because, if I'm not wrong, you can now create several tunnels, but then you cannot know how many you have or choose which one to stop.

A technical solution could be to store the port of each process in `tunnel.pid` instead of the timestamp, which is not used. And then show all of them with `status` and allow stopping a tunnel by port.